packages/block-editor/src/components/block-list/index.js

```diff
@@ -198,20 +198,22 @@ function Items( {
 				}
 
 				const selectedBlockClientId = getSelectedBlockClientId();
+				const isParentSelected = rootClientId === selectedBlockClientId;
+				const isEmpty = ! _order.length;
+				const shouldShowAppender = isParentSelected || isEmpty;
+
 				return {
 					order: _order,
 					selectedBlocks: getSelectedBlockClientIds(),
 					visibleBlocks: __unstableGetVisibleBlocks(),
 					shouldRenderAppender:
 						hasAppender &&
+						shouldShowAppender &&
 						__unstableGetEditorMode() !== 'zoom-out' &&
 						( hasCustomAppender
 							? ! getTemplateLock( rootClientId ) &&
 							  getBlockEditingMode( rootClientId ) !== 'disabled'
```

Comment:
This special condition for the custom appender is also very strange. If a template is locked or editing is disabled, default appenders should also be removed?

Reply:
This uses `rootClientId` so I wonder if we wanted to avoid a situation where blocks could be added if the parent was locked. Maybe @draganescu knows more.